db/migrate/20240716100102_update_student_lesson_summaries_with_enrollments.rb

```diff
@@ -0,0 +1,80 @@
+class UpdateStudentLessonSummariesWithEnrollments < ActiveRecord::Migration[7.1]
+  def up
+    drop_view :group_lesson_summaries
+    drop_view :lesson_table_rows
+
+    update_view :student_lesson_summaries, version: 7
+
+    create_view :group_lesson_summaries, version: 4
+    create_view :lesson_table_rows, version: 4
+    update_trigger
+  end
+
+  def down
+    drop_view :group_lesson_summaries
+    drop_view :lesson_table_rows
+
+    update_view :student_lesson_summaries, version: 6
+
+    create_view :group_lesson_summaries, version: 4
+    create_view :lesson_table_rows, version: 3
+    old_trigger
+  end
+
+  def update_trigger
+    execute <<~SQL
+      create or replace function update_enrollments()
+      returns trigger AS
+      $BODY$
+      declare
+        current_enrollment_group_id int := null;
+      BEGIN
+        -- Find the current group the student is enrolled in
+        SELECT group_id into current_enrollment_group_id FROM enrollments e where e.student_id = new.id and e.inactive_since is null;
+        -- Insert a new enrollment if this is the first time the student is being enrolled
+        if current_enrollment_group_id is null then
+            insert into enrollments (student_id, group_id, active_since, inactive_since, created_at, updated_at)
+            values (new.id, new.group_id, now(), null, now(), now());
+        else if current_enrollment_group_id != new.group_id then
```

Comment:
What you're doing here is nesting the `if` inside the `else` branch so you need 2 `end if` statements at the end.

This works but we can clean it up by using `elsif` which is the plpgsql way to do this